src/webgpu/error_test.ts

```diff
@@ -0,0 +1,103 @@
+import { Fixture } from '../common/framework/fixture.js';
+import { getGPU } from '../common/util/navigator_gpu.js';
+import { assert, raceWithRejectOnTimeout } from '../common/util/util.js';
+
+export class ErrorTest extends Fixture {
+  _device: GPUDevice | undefined = undefined;
+
+  get device(): GPUDevice {
+    assert(this._device !== undefined);
+    return this._device;
+  }
+
+  override async init(): Promise<void> {
+    await super.init();
+    const gpu = getGPU(this.rec);
+    const adapter = await gpu.requestAdapter();
+    assert(adapter !== null);
+
+    // We need to max out the adapter limits related to texture dimensions to more reliably cause an
+    // OOM error when asked for it, so set that on the device now.
+    const device = this.trackForCleanup(
+      await adapter.requestDevice({
+        requiredLimits: {
+          maxTextureDimension2D: adapter.limits.maxTextureDimension2D,
+          maxTextureArrayLayers: adapter.limits.maxTextureArrayLayers,
+        },
+      })
+    );
+    assert(device !== null);
+    this._device = device;
+  }
+
+  // Generates an error of the given filter type. For now, the errors are generated by calling a
+  // known code-path to cause the error. This can be updated in the future should there be a more
+  // direct way to inject errors.
```

Comment:
I know this is old but could you upgrade this to a doc comment?

Reply:
done